### ISSUE TITLE
Reveal is not closed by pressing Space

### ISSUE THREAD

**Issue by jmsfwk** [zurb/foundation-sites#9782]
Space does not trigger a Reveal to close on Firefox Developer Edition 53.0a2 (2017-02-11) (64-bit).

#### How to reproduce this bug:

1. Create a Reveal with a Close Button
2. Focus on the Close Button
3. Press <kbd>Space</kbd>

#### What should happen:

Reveal should close and return focus to the element focused before the Reveal was opened.

#### What happened instead:

The Reveal stays open and focus is returned to the element focused before the reveal was opened.

#### Test case:

https://codepen.io/jmsfwk/pen/dNrJKa

**Comment by dragthor** [comment]
Unsure about this request.  Fairly sure that Zurb 5.x did not have this behavior.

**Comment by jmsfwk** [comment]
When used with a callout Close Button is activated by <kbd>Space</kbd>.

https://codepen.io/jmsfwk/pen/ggJogd

**Comment by IamManchanda** [comment]
Yes callout's have these ..... i feel like this should be there in modal close too considering accessibility!

**Comment by kball** [comment]
@sethkane @Owlbertz as our accessibility experts, what do you think?

**Comment by sethkane** [comment]
It depends on how you are using the close element in the HTML.

If it is a <a href> then space bar does not natively using the space bar to trigger an event.
If you are using a <button> then the space bar has a native binding for that trigger.

The W3C has these guidelines for Modals https://www.w3.org/TR/wai-aria-practices/#dialog_modal

From the two examples given in this thread it appears that there might be some bad binding on the <button> elements inside of reveal.

FYI:
Buttons https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role space bar should trigger the event.

**Comment by kball** [comment]
@Owlbertz do you want to tackle this one?

**Comment by Owlbertz** [comment]
Should be solved via #9977.
